internal/schemautil/service.go

```
@@ -743,42 +746,140 @@ func copyConnectionInfoFromAPIResponseToTerraform(
-		props["database_name"] = connectionInfo.InfluxDBDatabaseName
 	case ServiceTypeKafka:
+		// KafkaHosts will be renamed to KafkaURIs in the next major version of the Go client.
+		// That's why we name the props key `uris` here.
+		props["uris"] = connectionInfo.KafkaHosts
 		props["access_cert"] = connectionInfo.KafkaAccessCert
 		props["access_key"] = connectionInfo.KafkaAccessKey
 		props["connect_uri"] = connectionInfo.KafkaConnectURI
 		props["rest_uri"] = connectionInfo.KafkaRestURI
 		props["schema_registry_uri"] = connectionInfo.SchemaRegistryURI
 	case ServiceTypePG:
+		// For compatibility with the old schema, we only set the first URI.
+		// TODO: Remove this block in the next major version. Keep `uris` key only, see below.
 		if connectionInfo.PostgresURIs != nil && len(connectionInfo.PostgresURIs) > 0 {
 			props["uri"] = connectionInfo.PostgresURIs[0]
 		}
+
+		props["uris"] = connectionInfo.PostgresURIs // Keep this instead of `uri` key.
+		props["bouncer"] = connectionInfo.PostgresBouncer
+
+		// For compatibility with the old schema, we only set the first params.
+		// TODO: Remove this block in the next major version. Keep `params` key only, see below.
 		if connectionInfo.PostgresParams != nil && len(connectionInfo.PostgresParams) > 0 {
 			params := connectionInfo.PostgresParams[0]
-			props["dbname"] = params.DatabaseName
+
 			props["host"] = params.Host
-			props["password"] = params.Password
+
 			port, err := strconv.ParseInt(params.Port, 10, 32)
 			if err == nil {
 				props["port"] = int(port)
 			}
+
 			props["sslmode"] = params.SSLMode
 			props["user"] = params.User
+			props["password"] = params.Password
+			props["dbname"] = params.DatabaseName
+
+		}
+
+		// Keep this instead of the above block.
+		params := make([]map[string]interface{}, len(connectionInfo.PostgresParams))
+		for i, p := range connectionInfo.PostgresParams {
+			port, err := strconv.ParseInt(p.Port, 10, 32)
+			if err != nil {
+				panic(err)
+			}
```

Please return the error.